osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs

```diff
@@ -102,16 +105,32 @@ public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
                                 soloScoreInfo.EndedAt = DateTime.SpecifyKind(score.ended_at!.Value, DateTimeKind.Utc);
                                 soloScoreInfo.BeatmapID = (int)score.beatmap_id;
 
-                                insertId = await db.InsertAsync(new SoloScore
+                                // InsertAsync is broken and doesn't support BIGINT primary keys...
+                                using (var insertCommand = db.CreateCommand())
                                 {
-                                    user_id = (int)score.user_id,
-                                    beatmap_id = (int)score.beatmap_id,
-                                    ruleset_id = item.ruleset_id,
-                                    preserve = true,
-                                    ScoreInfo = soloScoreInfo,
-                                    created_at = DateTime.SpecifyKind(score.created_at!.Value, DateTimeKind.Utc),
-                                    updated_at = DateTime.SpecifyKind(score.updated_at!.Value, DateTimeKind.Utc),
-                                });
+                                    insertCommand.CommandText = "INSERT INTO osu.solo_scores (user_id, beatmap_id, ruleset_id, data, preserve, created_at, updated_at) VALUES (@user_id, @beatmap_id, @ruleset_id, @data, @preserve, @created_at, @updated_at)";
+
+                                    var paramUserId = insertCommand.Parameters.Add("user_id", DbType.Int32);
+                                    var paramBeatmapId = insertCommand.Parameters.Add("beatmap_id", DbType.Int32);
+                                    var paramRulesetId = insertCommand.Parameters.Add("ruleset_id", DbType.Int32);
```

Comment:
Just for reference, the types here seem to be slightly different than the table definition. Not sure how much we care. I know that this successfully ran on production but just figure maybe best to bring it up when there's still a chance.

For reference:

```
mysql> describe solo_scores;
+------------+--------------------+------+-----+---------+----------------+
| Field      | Type               | Null | Key | Default | Extra          |
+------------+--------------------+------+-----+---------+----------------+
| id         | bigint unsigned    | NO   | PRI | NULL    | auto_increment |
| user_id    | int unsigned       | NO   | MUL | NULL    |                |
| beatmap_id | mediumint unsigned | NO   | MUL | NULL    |                |
| ruleset_id | smallint unsigned  | NO   |     | NULL    |                |
| data       | json               | NO   |     | NULL    |                |
| has_replay | tinyint(1)         | NO   |     | 0       |                |
| preserve   | tinyint(1)         | NO   | MUL | 0       |                |
| created_at | timestamp          | YES  |     | NULL    |                |
| updated_at | timestamp          | YES  |     | NULL    |                |
+------------+--------------------+------+-----+---------+----------------+
```

which I guess would amount to something like (warning, untested):

```diff
diff --git a/osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs b/osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs
index 7b3832a..9ea58a2 100644
--- a/osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs
+++ b/osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/MigratePlaylistScoresToSoloScoresCommand.cs
@@ -110,16 +110,16 @@ namespace osu.Server.Queues.ScoreStatisticsProcessor.Commands.Maintenance
                                 {
                                     insertCommand.CommandText = "INSERT INTO osu.solo_scores (user_id, beatmap_id, ruleset_id, data, preserve, created_at, updated_at) VALUES (@user_id, @beatmap_id, @ruleset_id, @data, @preserve, @created_at, @updated_at)";
 
-                                    var paramUserId = insertCommand.Parameters.Add("user_id", DbType.Int32);
-                                    var paramBeatmapId = insertCommand.Parameters.Add("beatmap_id", DbType.Int32);
-                                    var paramRulesetId = insertCommand.Parameters.Add("ruleset_id", DbType.Int32);
+                                    var paramUserId = insertCommand.Parameters.Add("user_id", DbType.UInt32);
+                                    var paramBeatmapId = insertCommand.Parameters.Add("beatmap_id", MySqlDbType.UInt24);
+                                    var paramRulesetId = insertCommand.Parameters.Add("ruleset_id", DbType.UInt16);
                                     var paramData = insertCommand.Parameters.Add("data", MySqlDbType.JSON);
                                     var paramPreserve = insertCommand.Parameters.Add("preserve", DbType.Boolean);
                                     var paramCreatedAt = insertCommand.Parameters.Add("created_at", MySqlDbType.Timestamp);
                                     var paramUpdatedAt = insertCommand.Parameters.Add("updated_at", MySqlDbType.Timestamp);
 
-                                    paramUserId.Value = (int)score.user_id;
-                                    paramBeatmapId.Value = (int)score.beatmap_id;
+                                    paramUserId.Value = score.user_id;
+                                    paramBeatmapId.Value = score.beatmap_id;
                                     paramRulesetId.Value = item.ruleset_id;
                                     paramPreserve.Value = true;
                                     paramData.Value = JsonConvert.SerializeObject(soloScoreInfo);

```

Reply:
Yeah I wasn't being too fussy, but since you've done the work feel free to apply. Might be useful in the future as a reference.